### PR TITLE
[Security] Skip user checks if not implementing UserInterface

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no 
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26871 
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php [hunk 1]
```diff
@@ -44,6 +45,10 @@ public function authenticate(TokenInterface $token)
             throw new AuthenticationException('Simple authenticator failed to return an authenticated token.');
         }
 
+        if ($authToken instanceof AnonymousToken) {
```
Comment (stof) [inline src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php]: the right check would be `!$user instanceof UserInterface`, to cover all cases where calling the user checker is not possible.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php (hunk 1 above)
Comment (chalasr) [reply src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php]: done

**Review by nicolas-grekas — APPROVED** [review]

**Review by aschempp — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Comment by leofeyer** [comment]
It seems that the changes have not been merged correctly into the 2.8 branch (and higher).

**2.7**
https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php#L50-L52

**2.8+**
https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php#L52-L67

I can confirm that the error no longer occurs in version 2.7, however, **it still does occur** in version 2.8+.

**Comment by nicolas-grekas** [comment]
But tests pass, so there is a missing test case.
@leofeyer would you like to submit a fix?

**Pull request merged**